go-controller/pkg/ovn/master.go

```diff
@@ -176,28 +174,38 @@ func (oc *Controller) upgradeToSingleSwitchOVNTopology(existingNodeList *kapi.No
 }
 
 func (oc *Controller) upgradeOVNTopology(existingNodes *kapi.NodeList) error {
-	// Find out the current OVN topology version, if "k8s-ovn-topo-version" external_ids is does not exist, it is pre
-	// OVN topology versioning, set version number to 0
+	// Find out the current OVN topology version. If "k8s-ovn-topo-version" key in external_ids column does not exist,
+	// it is prior to OVN topology versioning and therefore set version number to OvnCurrentTopologyVersion
 	ver := 0
-	stdout, stderr, err := util.RunOVNNbctl("get", "logical_router", util.OVNClusterRouter, "external_ids:k8s-ovn-topo-version")
+	stdout, stderr, err := util.RunOVNNbctl("--data=bare", "--no-headings", "--columns=name", "find", "logical_router",
+		fmt.Sprintf("name=%s", util.OVNClusterRouter))
 	if err != nil {
-		if strings.Contains(stderr, "no row") {
-			// no OVNClusterRouter exists, DB is empty, nothing to upgrade
-			ver = OvnCurrentTopologyVersion
-		} else {
-			klog.Infof("No version string found. The OVN topology is before versioning is introduced. Upgrade needed")
-		}
+		return fmt.Errorf("failed in retrieving %s to determine the current version of OVN logical topology: "+
+			"stderr: %q, error: %v", util.OVNClusterRouter, stderr, err)
+	}
+	if len(stdout) == 0 {
+		// no OVNClusterRouter exists, DB is empty, nothing to upgrade
```

Comment:
Then ver shoud be set to OvnCurrentTopologyVersion in this case, right?

Reply:
We return from here and then when the ovn_cluster_router is created later on, there we will set the external_ids:k8s-ovn-topo-version to current version.

Reply:
ok